### PR TITLE
chore(expiringcache): return additional bool from Get

### PR DESCRIPTION
**Pull request by janisz**
### Description

This PR is a prefactor for adding a types to expiringcache. As we cannot return typed `nil` we will need another indicator that value is nil thus we have a boolean to handle that.

- https://github.com/stackrox/stackrox/pull/11924
- https://github.com/stackrox/stackrox/pull/11961

- [x] CHANGELOG update is not needed
- [x] Documentation is not needed
### Testing
- [x] inspected CI results
#### Automated testing
- [x] modified existing tests
- [x] contributed **no automated tests**
#### How I validated my change
CI

### CODE REVIEW

File: central/networkgraph/flow/datastore/flow_impl.go [hunk 1]
```diff
@@ -84,7 +84,11 @@ func (fds *flowDataStoreImpl) adjustFlowsForGraphConfig(_ context.Context, flows
 }
 
 func (fds *flowDataStoreImpl) isDeletedDeployment(id string) bool {
-	deleted, _ := fds.deletedDeploymentsCache.Get(id).(bool)
+	v, ok := fds.deletedDeploymentsCache.Get(id)
+	if !ok {
+		return false
+	}
+	deleted, _ := v.(bool)
```
[the remark this answers is not in the document]
Comment (janisz) [reply central/networkgraph/flow/datastore/flow_impl.go:91]: this will be fixed in https://github.com/stackrox/stackrox/pull/11961

**Review by rhybrillou — APPROVED** [review]
I don't think it makes much sense functionally.
Nevertheless, I think it is technically possible to store nil values in the cache, and would then be reported as cache miss in the functional flows. The current caller locations probably do not differentiate between nil returned value as cache hit or cache miss.
LGTM, but for the sake of correctness, I'd return `(interface{}, bool)` in the functions serving `expiringCache.Get`.

File: pkg/expiringcache/cache.go [hunk 2]
```diff
@@ -98,21 +98,21 @@ func (e *expiringCacheImpl) addNoLock(key, value interface{}) {
 }
 
 // Get takes in a key and returns nil if the item doesn't exist or if the item has expired
-func (e *expiringCacheImpl) Get(key interface{}) interface{} {
+func (e *expiringCacheImpl) Get(key interface{}) (interface{}, bool) {
 	e.lock.Lock()
 	defer e.lock.Unlock()
 
 	e.cleanNoLock(e.clock.Now())
 	value := e.getValue(key)
```
[the remark this answers is not in the document]
Comment (janisz) [reply pkg/expiringcache/cache.go:106]: In theory yes but in our use cases we check for an non-nil object

**Comment by janisz** [comment]
@rhybrillou 
>I don't think it makes much sense functionally.

I agree but as mentioned in description it's prefactor for generic types and workaround for generic version as we cannot return `nil` in that case so you'll get an empty object instead. For example currently for a string cache we can return nil but with generic solution we need to return `""`.

**Pull request merged**